enterprise/backend/src/metabase_enterprise/advanced_config/file/settings.clj

```diff
@@ -14,4 +17,9 @@
   (log/info "Setting setting values from config file")
   (doseq [[setting-name setting-value] settings]
     (log/infof "Setting value for Setting %s" setting-name)
-    (setting/set! setting-name setting-value)))
+    (try
+      (setting/set! setting-name setting-value)
+      (catch ExceptionInfo e
+        (if (true? (:metabase.models.setting/unknown-setting-error (ex-data e)))
+          (log/warn (u/format-color :yellow "Ignoring unknown setting in config: %s." (name setting-name)))
+          (throw e))))))
```

Comment:
Why not just check if we can resolve the setting, and avoid throwing an exception in the first place?

Reply:
resolving the setting feels like an implementation detail of `setting/set!`, and anyway, resolving the setting with `resolve-setting` would throw the exact same exception. What are you specifically trying to avoid with this suggestion?

Reply:
Drat, I expected the semantics to be like `clojure.core/resolve` and to return nil.

My qualms with exceptions for control flow are basically that they're complex, and very expensive for the runtime. Checking a truthy value is much simpler than checking for a magic key and otherwise rethrowing, and it saves a whole bunch of reflection and serialization.

Why not just add an `metabase.models.setting/exists?` function?

Reply:
Good idea, done so in [e98f5cd](https://github.com/metabase/metabase/pull/43425/commits/e98f5cdc56c4067fa4ee97b8155e60595deb9064). I called it `registered?` though, not `exists?` to match the `registered-settings` name